### PR TITLE
Fixing issue #588

### LINKED ISSUE #588: Resolvers: Create tests for postsByOrganization.js
The Talawa-API code base needs to be 100% reliable. This means we need to have 100% test code coverage.

Tests need to be written for file `lib/resolvers/post_query/postsByOrganization.js`

- We will need the API to be refactored for all methods, classes and/or functions found in this file for testing to be correctly executed.
- When complete, all all methods, classes and/or functions in the refactored file will need to be tested. These tests must be placed in a
single file with the name `tests/resolvers/post_query/postsByOrganization.spec.js`. You may need to create the appropriate directory structure to do this.

### IMPORTANT:
Please refer to the parent issue on how to implement these tests correctly:

- https://github.com/PalisadoesFoundation/talawa-api/issues/490

### PR Acceptance Criteria

- When complete this file must show **100%** coverage when merged into the code base. This will be clearly visible when you submit your PR.
- [The current code coverage for the file can be found here](https://codecov.io/gh/PalisadoesFoundation/talawa-api/tree/develop/lib/resolvers/post_query/). If the file isn't found in this directory, or there is a 404 error, then tests have not been created.
- The PR will show a report for the code coverage for the file you have added. You can use that as a guide.

### PR DESCRIPTION
**Pull request by ebrahim354**
### bugfix 

Fixes #588

**Did you add tests for your changes?**

yes

**Summary**
Fixed a bug found on the postsByOrganization query resolver found after writing tests for the issue #588-

### CODE REVIEW

File: tests/resolvers/post_query/postsByOrganization.spec.js [hunk 1]
```diff
@@ -0,0 +1,590 @@
...
+const database = require('../../../db');
+const getUserId = require('../../functions/getUserId');
+const Organization = require('../../../lib/models/Organization');
+const Post = require('../../../lib/models/Post');
+const Comment = require('../../../lib/models/Comment');
+const User = require('../../../lib/models/User');
+
+// let mainToken;
+// let secondToken;
+// let thirdToken;
```
Comment (JamaicanFriedChicken) [inline tests/resolvers/post_query/postsByOrganization.spec.js:15]: if these variables aren't being used, it would be best to remove them.

Comment (ebrahim354) [reply tests/resolvers/post_query/postsByOrganization.spec.js:15]: I used them to initialize the seed data using the api then I faced a problem on adding imageUrl so used the database models directly and I'm not sure whether to leave it this way or change it so I just commented them if the database approach is ok with you i will remove them.

Comment (ebrahim354) [reply tests/resolvers/post_query/postsByOrganization.spec.js:15]: also the sorting problem I just solved it in place but I think a better way is to add a likeCount and commentCount property inside the user document to keep track of them directly from the database but  again I'm just waiting for feed back from you guys.

Comment (palisadoes) [reply tests/resolvers/post_query/postsByOrganization.spec.js:15]: Thanks, use the count properties.

Comment (palisadoes) [reply tests/resolvers/post_query/postsByOrganization.spec.js:15]: > I used them to initialize the seed data using the api then I faced a problem on adding imageUrl so instead I used the database models directly and I'm not sure whether to leave it this way or change it so I just commented them if the database approach is ok with you i will remove them.

What do you suggest as the best long term approach?

Comment (ebrahim354) [reply tests/resolvers/post_query/postsByOrganization.spec.js:15]: > > I used them to initialize the seed data using the api then I faced a problem on adding imageUrl so instead I used the database models directly and I'm not sure whether to leave it this way or change it so I just commented them if the database approach is ok with you i will remove them.
> 
> What do you suggest as the best long term approach?

I think using the database directly is better for performance ( better than using the whole api ) of tests which is beneficial with the increasing of tests added to the system, also the database approach solves the the problem I faced of adding imageUrl field without needing to pass unnecessary input validation etc..

Comment (palisadoes) [reply tests/resolvers/post_query/postsByOrganization.spec.js:15]: OK, please proceed. We need to fix this as many of our pushes to the repo are intermittently failing.

**Comment by palisadoes** [comment]
OK, please proceed with the DB,  `likeCount` and `commentCount` properties approaches. We need to fix this as many of our pushes to the repo are intermittently failing.

**Comment by ebrahim354** [comment]
> OK, please proceed with the DB, `likeCount` and `commentCount` properties approaches. We need to fix this as many of our pushes to the repo are intermittently failing.

done

**Review by palisadoes — APPROVED** [review]

**Pull request merged**